discovery-provider/src/tasks/entity_manager/entities/track.py

```diff
@@ -154,16 +154,14 @@ def is_valid_json_field(metadata, field):
     return False
 
 
-def populate_track_record_metadata(track_record, track_metadata, handle):
+def populate_track_record_metadata(track_record, track_metadata, handle, action):
     # Iterate over the track_record keys
     # Update track_record values for which keys exist in track_metadata
     track_record_attributes = track_record.get_attributes_dict()
     for key, _ in track_record_attributes.items():
         # For certain fields, update track_record under certain conditions
         if key == "is_unlisted":
-            # Only update `is_unlisted` if the track is unlisted. Once public, track cannot be
-            # made unlisted again
-            if "is_unlisted" in track_metadata and track_record.is_unlisted:
+            if "is_unlisted" in track_metadata:
```

Comment:
use whatever is in the metadata and validate in the validation step

Reply:
I think we can remove this case entirely now and have `is_unlisted` fall through to the else case